src/extension/background-script/actions/nostr/signEventOrPrompt.ts

```diff
@@ -74,6 +81,16 @@ const signEventOrPrompt = async (message: MessageSignEvent, sender: Sender) => {
           await addPermissionFor(permission, host, promptResponse.data.blocked);
         });
       }
+
+      if (promptResponse.data.confirm) {
```

Comment:
Why the same code again? (see below)

Reply:
i structured it the same as other prompts are strctured

if(hasPermission){
 calculate and return directly
}else{
	open prompt
	
	if prompt response is positive
		calculate and return directly
}

yup it can be improved but its in other prompts as well and can be fixed for all in one go as a separate pr. is it okay?

Reply:
Couldn't you just check for the confirm property and return if rejected? To me it seems the same code is being executed?

Reply:
when user cancels the permission they have option to store it as blocked permission. so we have to use same msg.reply() as we do for confirmed permissions. 

before the error was thrown when permission is cancelled from the screen itself. but with negative permissions and its permissions we need to pass aditional data.

so throwing errors and handling things are done in the action directly with the above mentioned approach.